### PR TITLE
Add Project Types, Stack Versioning and Upgrading to docs

### PR DESCRIPTION
**Pull request by knolleary**
Adds Project Types to the concepts.

Adds mention of Stack versioning to the managing stacks page - and on the Changing Stacks page.

Adds an upgrade guide to list any version-specific steps an admin must task as part of an upgrade.

### CODE REVIEW

File: docs/admin/README.md [hunk 1]
```diff
@@ -86,12 +86,29 @@ With the 0.1.0 release, the Teams page just lists the teams on the platform.
...
+each type has assigned to it, and how many projects have been created of that
+type.
+
 ### Managing Stacks
 
 The Stacks page can be used to manage the Project Stacks on the platform.
 
 It can be used to create and edit the stacks on the platform.
 
+With the 0.7 release, you can create a new version of an existing stack via the
```
Comment (sammachin) [inline docs/admin/README.md:107]: Should we mention version numbers in docs, feels like its just a way to go out of date, we have the release notes and blog post that shows when something was released.

**Review by hardillb — APPROVED** [review]

File: docs/admin/README.md (hunk 1 above)
Comment (knolleary) [inline docs/admin/README.md:107]:
```suggestion
You can create a new version of an existing stack via the
```

File: docs/user/changestack.md [hunk 2]
```diff
@@ -8,6 +8,10 @@ to upgrade Node-RED.
 **Note:** Stacks are created by Administrators and made available to the teams
 and users of the platform.
 
+With the 0.7 release of FlowForge, when an Administrator creates a new version
```
Comment (knolleary) [inline docs/user/changestack.md:11]:
```suggestion
When an Administrator creates a new version
```

**Comment by knolleary** [comment]
Have removed the version numbers as suggested.

**Pull request merged**